<a href="https://colab.research.google.com/github/Verona1901/mat_stat/blob/main/DZ10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов.
Даны значения роста в трех группах случайно выбранных спортсменов:

Футболисты: 173, 175, 180, 178, 177, 185, 183, 182.

Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180.

Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [59]:
import numpy as np
from scipy import stats

In [60]:
football_players = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hockey_players = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
weightlifters = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

α = 0,05. Проверим данные на нормальность распределения с помощью теста Шапиро-Уилка

In [61]:
stats.shapiro(football_players)

ShapiroResult(statistic=0.9775082468986511, pvalue=0.9495404362678528)

In [62]:
stats.shapiro(hockey_players)

ShapiroResult(statistic=0.9579196572303772, pvalue=0.7763139009475708)

In [63]:
stats.shapiro(weightlifters)

ShapiroResult(statistic=0.9386808276176453, pvalue=0.5051165223121643)

Значение p-value > alpha - принимается гипотеза H0, выборка подчиняется нормальному закону.

Выполним тест Бартлетта для определения того, равны ли дисперсии между группами спортсменов.
α = 0,05

In [64]:
stats.bartlett(football_players, hockey_players, weightlifters)

BartlettResult(statistic=0.4640521043406442, pvalue=0.7929254656083131)

Так как p-value больше уровня значимости  α = 0,05, Нулевая гипотеза не отвергается,  дисперсия среди каждой группы равна.

Таким образом, можно выполнить однофакторный дисперсионный анализ.

Сформулируем гипотезы:

H 0 (нулевая гипотеза): µ 1 = µ 2 = µ 3 (различий среднего роста среди взрослых футболистов, хоккеистов и штангистов нет)

H 1 (альтернативная гипотеза): по крайней мере одно среднее значение отличаетсяот остальных. имеются различия среднего роста среди спортсменов.

Выполним односторонний ANOVA.

In [65]:
stats.f_oneway(football_players, hockey_players, weightlifters)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Статистика F-теста = 5.50005, а соответствующее значение p = 0.01048.Поскольку p-значение меньше 0,05, отвергаем нулевую гипотезу. Имеются различия среднего роста спортсменов.

Проведем post hoc tests для определения в каких группах имеется различие среднего роста.

In [66]:
stats.f_oneway(football_players, hockey_players)

F_onewayResult(statistic=0.038473767885532584, pvalue=0.8471274829862326)

In [67]:
stats.f_oneway(football_players, weightlifters)

F_onewayResult(statistic=9.232797542249022, pvalue=0.007418502503470366)

In [68]:
stats.f_oneway(hockey_players, weightlifters)

F_onewayResult(statistic=6.86, pvalue=0.017384077963160147)

Таким образом, можно сделать вывод, что в группах спортсменов (football_players, hockey_players) - футболистов и хоккеистов, можно считать, что статистически значимых различий нет, так как pvalue=0.84712 > 0,05. 

Различия наблюдаются только между футболистами и штангистами и хоккеистами и штангистами.